ambry-replication/src/main/java/com.github.ambry.replication/ReplicaThread.java

```diff
@@ -229,20 +219,79 @@ public void run() {
   }
 
   /**
-   * Replicas from the given replicas
-   * @param replicasToReplicate list of {@link RemoteReplicaInfo} by data node
+   * Remove {@link RemoteReplicaInfo} from current {@link ReplicaThread}.
+   * @param remoteReplicaInfo {@link RemoteReplicaInfo} to remove.
+   */
+  void removeRemoteReplicaInfo(RemoteReplicaInfo remoteReplicaInfo) {
+    lock.lock();
+    try {
+      DataNodeId dataNodeId = remoteReplicaInfo.getReplicaId().getDataNodeId();
+      Set<RemoteReplicaInfo> remoteReplicaInfos = replicasToReplicateGroupedByNode.get(dataNodeId);
+      if (remoteReplicaInfos != null) {
+        if (!remoteReplicaInfos.remove(remoteReplicaInfo)) {
```

Comment:
In this remove replica method, do we need to update `allReplicatedPartitions` as well? If yes, we have to convert `allReplicatedPartitions` to a Map<PartitionId, List<ReplicaId>>.

Reply:
Thanks for pointing out this. 
there is some overlap between disable partition and remove replica, but there is also conflict: one by partition, the other by replica. 
It's not a small change to solve this problem. So I would like do it in follow up PRs. 
But the code is good because we:
1. Don't do dynamically add/remote replica to ReplicationManager.
2. Don't disable/enable partition to CloudBackUpManager.